src/main/scala/uk/gov/homeoffice/drt/export/zip/ZipFilesInDirectory.scala

```diff
@@ -0,0 +1,85 @@
+package uk.gov.homeoffice.drt.export.zip
+
+import java.io.File
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.zip.ZipFile
+
+import grizzled.slf4j.Logging
+import org.apache.commons.io.IOUtils
+import uk.gov.homeoffice.drt.export.HasConfig
+
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+trait ZipFilesInDirectory extends Logging with HasConfig {
+
+  val dqRegex: Regex = "drt_dq_([0-9]{2})([0-9]{2})([0-9]{2})_([0-9]{6})_([0-9]{4})\\.zip".r
+
+  val directory: String = config.getString("zipDirectory")
+
+  val allowAllEvents = config.getBoolean("allowAllEvents")
+
+  def getFilesInDirectory(fileNameOption : Option[String] = None): List[String] = {
+    val d = new File(directory)
+
+    if (!d.exists && !d.isDirectory) {
+      throw new IllegalArgumentException(s"Directory $directory should exist and be a directory.")
+    }
+    val files = d.listFiles.filter(f => f.isFile && f.getName.matches(dqRegex.regex)).sortBy(f => f.getName match {
+      case dqRegex(year, month, day, second, third) => (year.toInt, month.toInt, day.toInt, second.toInt, third.toInt)
+      case _ => (-1, -1 , -1, -1, -1)
```

Comment:
should we maybe do a collect to filter out the wrongly named files first? There are a few in there that are test files.

Reply:
you'd then be able to just sort by filename I think.